editor/src/clj/editor/resource.clj

```diff
@@ -214,8 +270,16 @@
   (resource-name [this] name)
   (workspace [this] workspace)
   (resource-hash [this] (hash (proj-path this)))
-  (openable? [this] (= :file source-type))
-  (editable? [this] editable)
+  (openable? [this]
+    (and (= :file source-type)
+         (if (:editor-openable (resource-type this))
+           (loaded? this)
+           true)))
```

Comment:
This may look a bit weird, but remember we only care if we're loaded or not if we're opening it in the Defold editor - it's perfectly safe to open unloaded files in external editors.